lark/lexer.py

```diff
@@ -9,12 +10,23 @@
 ###{standalone
 from copy import copy
 
+from types import ModuleType
+from typing import (
+    TypeVar, Type, Tuple, List, Dict, Iterator, Collection, Callable, Optional, FrozenSet, Any,
+    Pattern as REPattern, TYPE_CHECKING
+)
 
-class Pattern(Serialize):
-    raw = None
-    type = None
+if TYPE_CHECKING:
+    from .common import LexerConf
```

Comment:
Probably a TODO for later, but I think we can instead of guarding the circle with `TYPE_CHECKING` move more stuff into `common.py`. That is why it exists. ( In this case, the definition of `TerminalDef`/the Patterns/`Token`)